### ISSUE TITLE
question: does Aries require a blockchain?

### ISSUE THREAD

**Issue by bobpretzel** [hyperledger/aries-rfcs#352]
"Hyperledger" implies a fast distributed ledger. Is there anything in Aries which would preclude use of, say, a hashgraph, DAG or tangle to implement a fast, scalable, efficient distributed ledger? The Nakamoto Proof-of-Work blockchain was revolutionary in its time but alternatives have emerged which might help to avoid the appearance of multiple SSI silos due to limitations in the underlying ledger.

**Comment by TelegramSam** [comment]
The overall vision of the project is that use of an Aries agent does not require any particular ledger. In the short term, this is enabled by the use of Peer DIDs, which are not recorded on any ledger. (You could argue that they maintain a ledger between the peers...) 

Many portions of the Aries vision are enabled via the use of an Oracle, which ledgers do provide. Use of common Schemas, etc. rely on this to some degree. Those portions of Aries are likely to continue to require access to those sources.

To summarize, the answer is Not strictly, but probably in a practical sense.

**Comment by swcurran** [comment]
I don't think that `did:peer` DIDs are "in the short term".  They are a crucial component (IMHO) to the Aries model. They are by definition not appropriate for globally resolvable DIDs. The current expectation is that globally resolvable DIDs do require some sort of public ledger implementation to be decentralized. Hence, blockchain.  If you can achieve the decentralized goal with blockchain, then blockchain would not be needed.

Not that it's quite possible to have Aries agents that operate without using things that rely on globally resolvable DIDs (e.g. without Verifiable Credentials). If such things are not needed, an Aries agent would not need access to a ledger.

That said, if another agent happened to use a globally resolvable DID, the "ledger-less" Aries agent would probably need to resolve that DID...

**Comment by bobpretzel** [comment]
ok - the appearance of the term "blockchain-rooted" just struck me as unusual. I thought, perhaps incorrectly, that a pluggable interface for a distributed, public, permissioned ledger might appear as a (possibly thin) interface for a pluggable KV store with sufficient consistency/ordering/.. guarantees to meet Aries' needs.

This isn't intended as a criticism. Identity is at the foundation of human rights and the ability to scale has merit since those most in need of identity often cannot rely on their nation to provide it during crises. It seemed worthwhile to raise the issue - hopefully it's constructive.

**Comment by swcurran** [comment]
Blockchain entered the conversation for decentralization reasons, not for payment. When there is a central registry of identity information, it becomes a target for attack to retrieve related information, and it gives a central authority control to remove entries as they feel. In the decentralized model, we want to give the user control over how their identity attributes are used. The sources of information about your identity are not centralized but rather are retained with their respective issuing authorities. Credentials are issued to you in a manner such that you can prove them as you need to complete transactions - exposing no less information than is needed, and importantly, no more than is needed. Entities needing to know things about you get that information from you, and don't have to connect with the issuer via an integration to get that information.

There is a lot to this topic. Looking quickly for a source on why decentralized ledgers vs. a database, I think this is a good place to start - https://w3c.github.io/did-core/#introduction, and follow the links within the section for more details about the benefit of this approach.

**Comment by swcurran** [comment]
Reviewed on the Aries Working Group call 20200108. We think this issue is sufficiently answered and we can close it. Please reopen if you disagree.
